code/modules/antagonists/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,1093 @@
+				enthralling = FALSE
+				return
+		enthralling = FALSE
+		to_chat(user, "<span class='shadowling'>You have enthralled <b>[target.real_name]</b>!</span>")
+		target.visible_message("<span class='big'>[target] looks to have experienced a revelation!</span>", \
+							   "<span class='warning'>False faces all d<b>ark not real not real not--</b></span>")
+		target.setOxyLoss(0) //In case the shadowling was choking them out
+		target.mind.special_role = "thrall"
+		var/obj/item/organ/internal/shadowtumor/ST = new
+		ST.Insert(target, FALSE, FALSE)
+		target.add_thrall()
+		if(target.reagents.has_reagent(/datum/reagent/consumable/frostoil)) //Stabilize body temp incase the shadowling froze them earlier
+			target.reagents.remove_reagent(/datum/reagent/consumable/frostoil)
+			to_chat(target, "<span class='notice'>You feel warmer... It feels good.</span>")
+			target.bodytemperature = 310
+
+/obj/effect/proc_holder/spell/self/shadowling_hivemind //Lets a shadowling talk to its allies
+	name = "Hivemind Commune"
+	desc = "Allows you to silently communicate with all other shadowlings and thralls."
+	panel = "Shadowling Abilities"
+	charge_max = 0
+	human_req = TRUE
+	clothes_req = FALSE
+	action_icon = 'icons/mob/actions.dmi'
+	action_icon_state = "commune"
+
+/obj/effect/proc_holder/spell/self/shadowling_hivemind/cast(mob/living/user,mob/user = usr)
+	if(!IS_SHADOW(user))
+		to_chat(user, "<span class='warning'>You must be a shadowling to do that!</span>")
+		return
+	var/text = stripped_input(user, "What do you want to say your thralls and fellow shadowlings?.", "Hive Chat", "")
+	if(!text)
+		return
+	var/my_message = "<span class='shadowling command_headset'><b>\[Shadowling\]</b><i> [user.real_name]</i>: [text]</span></font>"
+	for(var/mob/M in GLOB.mob_list)
+		if(IS_SHADOW_OR_THRALL(M))
+			to_chat(M, my_message)
+		if(M in GLOB.dead_mob_list)
+			to_chat(M, "<a href='?src=[REF(M)];follow=[REF(user)]'>(F)</a> [my_message]")
+	log_say("[user.real_name]/[user.key] : [text]")
+
+/obj/effect/proc_holder/spell/self/shadowling_regenarmor //Resets a shadowling's species to normal, removes genetic defects, and re-equips their armor
+	name = "Rapid Re-Hatch"
+	desc = "Re-forms protective chitin that may be lost during cloning or similar processes."
+	panel = "Shadowling Abilities"
+	charge_max = 600
+	human_req = TRUE
+	clothes_req = FALSE
+	action_icon = 'icons/mob/actions.dmi'
+	action_icon_state = "regen_armor"
+
+/obj/effect/proc_holder/spell/self/shadowling_regenarmor/cast(mob/living/carbon/human/user)
+	if(!IS_SHADOW(user))
+		to_chat(user, "<span class='warning'>You must be a shadowling to do this!</span>")
+		revert_cast()
+		return
+	user.visible_message("<span class='warning'>[user]'s skin suddenly bubbles and shifts around their body!</span>", \
+						 "<span class='shadowling'>You regenerate your protective armor and cleanse your form of defects.</span>")
+	user.setCloneLoss(0)
```

Comment:
I really don't see why shadowlings would need cloneloss immunity considering cloning is gone and the biggest source of it is the abductor gun

Reply:
Imagine a shadowling going into medbay and asking "uhhh can you let me into a cryotube i need to heal some cloneloss"

Reply:
I don't need to imagine it because the odds are stupid low that they will meet an abductor, and they might aswell enthrall medbay staff for it. That's not excuse

Reply:
They can meet a slime!